### PR TITLE
samples: Merge Azure IoT Hub and Azure FOTA samples

### PR DESCRIPTION
**Pull request by jtguggedal**
The azure_fota sample is removed and the FOTA functionality is
now shown in the azure_iot_hub sample instead.
This is done because there was a significant overlap in the
sample content and most applications will most likely want to
implement both IoT Hub communication and FOTA together.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -411,6 +411,10 @@ Networking samples
   * Added support for Wi-Fi and LTE connectivity through the connection manager API.
   * Moved the sample from :file:`nrf9160/azure_iot_hub` folder to :file:`net/azure_iot_hub`.
     The documentation is now found in the :ref:`networking_samples` section.
+  * Added FOTA support using :ref:`azure_fota`.
+
+* Removed Azure FOTA sample.
+  FOTA using Azure IoT Hub is now demonstrated in :ref:`azure_iot_hub`.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:417]:
```suggestion
  FOTA using Azure IoT Hub is now demonstrated in the :ref:`azure_iot_hub` sample.
```

File: samples/net/azure_iot_hub/README.rst [hunk 2]
```diff
@@ -123,12 +146,37 @@ Optionally, follow the instructions at `Azure IoT Explorer`_ to install and conf
...
+	            "path": "firmware/my-app-update.bin"
+	        },
+	        "fwFragmentSize": 1800
+	    }
+	}
+
+#. Apply the changes to the device twin by clicking :guilabel:`Save`. Updating the device twin causes a message to be sent to the device, containing the updates to the *desired* object.
+#. In the terminal emulator, observe that the new firmware image is downloaded and installed as shown in :ref:`sampoutput_azure_iot`.
+#. When the development kit reboots, observe that the sample displays the logs from the new application.
+#. In `Azure IoT Explorer`_ or Azure Portal, confirm that the device twin contains the updated application version in the ``reported.firmware.fwVersion`` field.
```
Comment (peknis) [inline samples/net/azure_iot_hub/README.rst:172]:
```suggestion
#. Confirm that the device twin contains the updated application version in the ``reported.firmware.fwVersion`` field in `Azure IoT Explorer`_ or `Azure Portal`_.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Comment by peknis** [comment]
Looks like the Azure FOTA sample doc has been linked to from a few other docs as well (my search for :ref: `azure_fota_sample` resulted in six hits). Please fix those as well.

**Review by simensrostad — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -411,6 +411,10 @@ Networking samples
   * Added support for Wi-Fi and LTE connectivity through the connection manager API.
   * Moved the sample from :file:`nrf9160/azure_iot_hub` folder to :file:`net/azure_iot_hub`.
     The documentation is now found in the :ref:`networking_samples` section.
+  * Added FOTA support using :ref:`azure_fota`.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:414]:
```suggestion
  * Added FOTA support using the :ref:`lib_azure_fota` library.
```

File: samples/net/azure_iot_hub/README.rst [hunk 4]
```diff
@@ -57,6 +69,7 @@ Setup
 For the sample to work as intended, you must set up and configure an Azure IoT Hub instance.
 See :ref:`configure_options_azure_iot` for information on the configuration options that you can use to create an Azure IoT Hub instance.
 Also, for a successful TLS connection to the Azure IoT Hub instance, the device needs to have certificates provisioned.
+If you want to test FOTA, ensure that also the required credentials for the file server are provisioned and that the :kconfig:option:`CONFIG_AZURE_FOTA_SEC_TAG` Kconfig option is set accordingly.
```
Comment (peknis) [inline samples/net/azure_iot_hub/README.rst:72]:
```suggestion
If you want to test FOTA, ensure that also the required credentials for the file server are provisioned and the :kconfig:option:`CONFIG_AZURE_FOTA_SEC_TAG` Kconfig option is set accordingly.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/net/azure_iot_hub/README.rst [hunk 5]
```diff
@@ -123,12 +146,38 @@ Optionally, follow the instructions at `Azure IoT Explorer`_ to install and conf
...
+
+   .. parsed-literal::
+      :class: highlight
+
+	{
+	    "firmware": {
+	        "fwVersion": "v0.0.2-dev",
+	        "jobId": "ca186d4b-4171-4209-a49d-700c35567d1d",
+	        "fwLocation": {
+	            "host": "my-fota-server.comt",
```
Comment (coderbyheart) [inline samples/net/azure_iot_hub/README.rst:162]:
```suggestion
	            "host": "example.com",
```

Use generic example domain.

File: samples/net/azure_iot_hub/README.rst [hunk 6]
```diff
@@ -123,12 +146,38 @@ Optionally, follow the instructions at `Azure IoT Explorer`_ to install and conf
...
+	            "path": "firmware/my-app-update.bin"
+	        },
```
Comment (coderbyheart) [inline samples/net/azure_iot_hub/README.rst:164]: **question:** can't we use a fully qualified URL by now? AFAIR the download client is now able to use that?

File: samples/net/azure_iot_hub/README.rst [hunk 7]
```diff
@@ -123,12 +146,38 @@ Optionally, follow the instructions at `Azure IoT Explorer`_ to install and conf
...
+#. Observe that the sample displays the logs from the new application when the development kit reboots.
+#. Confirm that the device twin contains the updated application version in the ``reported.firmware.fwVersion`` field in `Azure IoT Explorer`_ or `Azure Portal`_.
+
 .. _sampoutput_azure_iot:
 
 Sample output
 =============
 
-When the sample runs, the device boots, and the sample displays the following output in the terminal over UART:
+When the sample runs, the device boots, and the sample displays the following output in the terminal over UART on an nRF7002 DK:
```
Comment (coderbyheart) [inline samples/net/azure_iot_hub/README.rst:180]: *question:* Why is `nRF7002` relevant?

File: samples/net/azure_iot_hub/README.rst [hunk 8]
```diff
@@ -161,13 +211,37 @@ If a new telemetry interval is set in the device twin, the console output is lik
 	<inf> azure_iot_hub_sample: Event was successfully sent
 	<inf> azure_iot_hub_sample: Next event will be sent in 60 seconds
 
+If a new FOTA update is initiated, the console output is like this:
+
+.. code-block:: console
+
+	<inf> azure_iot_hub_sample: AZURE_IOT_HUB_EVT_TWIN_RESULT_SUCCESS, ID: 140
+	<inf> azure_fota: Attempting to download firmware (version 'v0.0.2-dev') from my-fota-server.com/firmware/app_update.bin
```
Comment (coderbyheart) [inline samples/net/azure_iot_hub/README.rst:219]:
```suggestion
	<inf> azure_fota: Attempting to download firmware (version 'v0.0.2-dev') from example.com/firmware/app_update.bin
```

File: samples/net/azure_iot_hub/README.rst [hunk 9]
```diff
@@ -161,13 +211,37 @@ If a new telemetry interval is set in the device twin, the console output is lik
...
+	<inf> download_client: Downloading: firmware/app_update.bin [0]
+	<inf> azure_iot_hub_sample: AZURE_IOT_HUB_EVT_FOTA_START
+	<inf> azure_iot_hub_sample: AZURE_IOT_HUB_EVT_TWIN_DESIRED_RECEIVED
+	<inf> download_client: Setting up TLS credentials, sec tag count 1
+	<inf> download_client: Connecting to my-fota-server.com
```
Comment (coderbyheart) [inline samples/net/azure_iot_hub/README.rst:224]:
```suggestion
	<inf> download_client: Connecting to example.com
```

**Review by coderbyheart — CHANGES_REQUESTED** [review]

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/net/azure_iot_hub/README.rst [hunk 10]
```diff
@@ -80,6 +93,16 @@ As an example, the following compiles with DPS for nRF9160DK:
 * :kconfig:option:`CONFIG_AZURE_IOT_HUB_DPS_REG_ID` - Sets the Azure IoT Hub DPS registration ID. It can be provided at run time. By default, the sample uses the device ID as the registration ID and sets it at run time.
 * :kconfig:option:`CONFIG_AZURE_IOT_HUB_DPS_ID_SCOPE` - Sets the DPS ID scope of the Azure IoT Hub. This can be provided at run time.
 
+For FOTA, check and configure the following library Kconfig options:
+
+* :kconfig:option:`CONFIG_AZURE_FOTA_APP_VERSION` - Defines the application version string. Indicates the current firmware version on the development kit.
+* :kconfig:option:`CONFIG_AZURE_FOTA_APP_VERSION_AUTO` - Automatically generates the application version. If enabled, :kconfig:option:`CONFIG_AZURE_FOTA_APP_VERSION` is ignored.
+* :kconfig:option:`CONFIG_AZURE_FOTA_TLS` - Enables HTTPS for downloads. By default, TLS is enabled and currently, the transport protocol must be configured at compile time.
+* :kconfig:option:`CONFIG_AZURE_FOTA_SEC_TAG` - Sets the security tag for TLS credentials when using HTTPS as the transport layer. See :ref:`certificates` for more details.
```
[the remark this answers is not in the document]
Comment (KajaKoren) [reply samples/net/azure_iot_hub/README.rst:101]: yes, it seems to reference the old README file of the Azure Fota sample.. I can move the certification text to this sample instead.

[the remark this answers is not in the document]
Comment (KajaKoren) [reply samples/net/azure_iot_hub/README.rst:101]: or refer to the `prereq_connect_to_azure_iot_hub` instead, is being referred to in the credential reference anyway.

File: doc/nrf/releases_and_maturity/releases/release-notes-2.3.0.rst [hunk 11]
```diff
@@ -554,7 +554,7 @@ nRF9160 samples
     * Removed user prompt and button press handling for FOTA enablement.
     * The sample now uses a partition in external flash for full modem FOTA updates.
 
-* :ref:`azure_fota_sample` sample:
+* azure_fota_sample sample:
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-2.3.0.rst:557]:
```suggestion
* nRF9160: Azure FOTA sample:
```

Comment (KajaKoren) [reply doc/nrf/releases_and_maturity/releases/release-notes-2.3.0.rst]: Fixed

**Review by peknis — APPROVED** [review]

**Review by coderbyheart — APPROVED** [review]

**Comment by KajaKoren** [comment]
A change added due to this change in azure-fota sample: [samples: cellular: azure_fota: use default C library](https://github.com/nrfconnect/sdk-nrf/commit/28c2e5f82b9103c82bc66f470d85183d0f5b4eee)

**Review by KajaKoren — APPROVED** [review]

**Pull request merged**